storage-node/packages/runtime-api/index.js

```diff
@@ -127,6 +127,31 @@ class RuntimeApi {
     })
   }
 
+  // Get cached nonce and use unless system nonce is greater, to avoid stale nonce if
+  // there was a long gap in time between calls to signAndSend during which an external app
+  // submitted a transaction.
+  async selectBestNonce(accountId) {
+    const cachedNonce = this.nonces[accountId]
+    // In future use this rpc method to take the pending tx pool into account when fetching the nonce
+    // const nonce = await this.api.rpc.system.accountNextIndex(accountId)
+    const systemNonce = await this.api.query.system.accountNonce(accountId)
+
+    if (cachedNonce) {
```

Comment:
Seems like this could be reduced to
`return (cachedNonce && cachedNonce.gte(systemNonce)) ? cachedNonce : systemNonce`
But it's pretty much a non-issue

Reply:
Yes I started off like that, but then decided to be more verbose for clarity and so I can add some comments :)